src/main/java/com/iexec/worker/docker/DockerService.java

```diff
@@ -202,6 +187,80 @@ public String getIexecOutBind(String chainTaskId) {
                 IexecFileHelper.SLASH_IEXEC_OUT;
     }
 
+    /**
+     * Stop all running containers launched by the worker via this current service
+     * and remove them from running containers record. The container itself is not
+     * removed here as it is removed by its watcher thread.
+     */
+    public void stopAllRunningContainers() {
+        log.info("About to stop all running containers [runningContainers:{}]",
+                runningContainersRecord);
+        List.copyOf(runningContainersRecord)
+                .forEach(containerName -> stopRunningContainer(containerName));
+    }
+
+    /**
+     * Stop running containers related to a certain task and remove them from the
+     * running containers record. The container can be pre-compute, compute, or
+     * post-compute. This is needed when the worker aborts a task. The container itself
+     * is not removed here as it is removed by its watcher thread.
+     * 
+     * @param chainTaskId
+     */
+    public void stopTaskRunningContainers(String chainTaskId) {
+        log.info("Cleaning task containers [chainTaskId:{}]", chainTaskId);
+        runningContainersRecord.stream()
+                .filter(containerName -> containerName.contains(chainTaskId))
```

Comment:
:)

Reply:
Maybe we prefer something like using `stopRunningContainersHavingPattern(..)` and calling it from a task service, to avoid introducing task concept in this Docker service.